network/p2p/inspector/validation/control_message_validation_inspector.go

```diff
@@ -209,6 +209,23 @@ func (c *ControlMsgValidationInspector) Inspect(from peer.ID, rpc *pubsub.RPC) e
 		return nil
 	}
 
+	// check peer identity when running private network
+	// sanity check: rpc inspection should be disabled on public networks
+	if c.networkingType == network.PrivateNetwork {
```

Comment:
Please also add a runtime flag switch to enable/disable this when needed:
```go
c.networkingType == network.PrivateNetwork && <switchIsOn> {
  // rest of logic
}
```

Reply:
https://github.com/onflow/flow-go/pull/5449/commits/37103653eded1190d4b96c4e4d9dbed6f75a2712